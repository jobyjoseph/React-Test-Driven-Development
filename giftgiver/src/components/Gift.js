import React from "react";
import { Form, FormGroup, FormControl, ControlLabel, Button, FormLabel } from "react-bootstrap";

class Gift extends React.Component {

  state = {
    person: "",
    present: ""
  }

  onPersonChange = (e) => {
    this.setState({
      person: e.target.value
    });
  }

  render() {
    return (
      <div>
        <Form>
          <FormGroup>
            <FormLabel>Person</FormLabel>
            <FormControl className="input-person" onChange={this.onPersonChange}/>
          </FormGroup>
        </Form>
      </div>
    );
  }
}

export default Gift;