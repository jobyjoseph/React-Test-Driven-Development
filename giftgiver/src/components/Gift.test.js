import React from "react";
import { shallow } from "enzyme";
import Gift from "./Gift";

describe("Gift", () => {

  const mockRemove = jest.fn();
  const id = 1;
  const props = { gift: {id}, removeGift: mockRemove };
  const gift = shallow(<Gift {...props}/>);

  it("Component renders correctly", () => {
    expect(gift).toMatchSnapshot();
  });

  it("initializes a person and present in `state`", () => {
    expect(gift.state()).toEqual({person:"", present:""})
  })

  describe("When typing into the person input", () => {
    const person = "Uncle";

    beforeEach(() => {
      gift.find('.input-person').simulate("change", {target: {value: person}});
    });

    it("updates the person in `state`", () => {
      expect(gift.state().person).toEqual(person);
    });
  });

  describe("When typing into the present input", () => {
    const present = "Wallet";

    beforeEach(() => {
      gift.find(".input-present").simulate("change", {target: {value: present}});
    });

    it("updates the present in `state`", () => {
      expect(gift.state().present).toEqual(present);
    });

  });

  describe("When `Remove Gift` button is clicked", () => {
    beforeEach(() => {
      gift.find(".btn-remove").simulate("click");
    });

    it("calls removeGift callback", () => {
      expect(mockRemove).toHaveBeenCalledWith(id);
    })
  });

});