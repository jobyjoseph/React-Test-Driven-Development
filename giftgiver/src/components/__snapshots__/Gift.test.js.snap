// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Component renders correctly 1`] = `
<div>
  <Form
    inline={false}
  >
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Person
      </FormLabel>
      <FormControl
        className="input-person"
        onChange={[Function]}
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Present
      </FormLabel>
      <FormControl
        className="input-present"
        onChange={[Function]}
      />
    </FormGroup>
  </Form>
  <Button
    active={false}
    className="btn-remove"
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Remove Gift
  </Button>
</div>
`;
